### ISSUE TITLE
Netease broken again

### ISSUE THREAD

**Issue by theli-ua** [rg3/youtube-dl#8239]
Tests in master for extractor will fail.
Looks like they switched download server again.

**Comment by theli-ua** [comment]
Note, if you switch to m2 and switch to using some chinese DNS server it works just fine.
Support for that could be implemented (for downloads using youtube-dl) using pydns/dnspython. But I doubt that youtube-dl would want such dependency.

Users could also set their resolvers to use one of chinese public DNS servers or even just pu2 m2.music.126.net into their hosts file.

In any case - current m5.music.126.net server hardcoded into youtube-dl doesn't work

**Comment by yan12125** [comment]
Which URL are you trying to download?

**Comment by theli-ua** [comment]
@yan12125 
eg the one from the tests http://music.163.com/#/song?id=32102397  will get you http://m5.music.126.net/Hj8j3B9wn2TetQue7VJ-iQ==/2901611187169450.mp3  and end up with 404.
So far I've been getting 404 for all links

**Comment by yan12125** [comment]
When I click play in http://music.163.com/#/song?id=32102397, it says that I have to pay for this song to listen to it:
```
唱片公司要求，当前资源须付费使用
```
Which browser are you using? Are you inside Mainland China?

**Comment by theli-ua** [comment]
I haven't even tried to open it in browser. I just tried to download it using youtube-dl . Its one of the test urls inside extractor/neteasemusic.py 
No, I'm not in China

**Comment by yan12125** [comment]
If a video/song can't be played in browsers, there's nothing much youtube-dl can do. Test cases may become invalid as time goes by. It's possible this song is once free but later requires payment.

**Comment by theli-ua** [comment]
Sure, but the sond could be downloaded from "http://m2.music.126.net/zUA4iqTZyZFncNkFNUwAjw==/5991238859813420.mp3" 
Do you have an example of url that works with youtube-dl?

**Comment by yan12125** [comment]
Have you changed your DNS settings? m2.music.126.net is resolved to 1.1.1.1, which is apparently not a NetEase IP.
```
$ curl -v http://m2.music.126.net/Hj8j3B9wn2TetQue7VJ-iQ\=\=/2901611187169450.mp3
*   Trying 1.1.1.1...
* connect to 1.1.1.1 port 80 failed: Connection refused
* Failed to connect to m2.music.126.net port 80: Connection refused
* Closing connection 0
curl: (7) Failed to connect to m2.music.126.net port 80: Connection refused
```
